contracts/TokenVesting.sol

```diff
@@ -2,355 +2,442 @@
-    * @return the vesting schedule structure information
-    */
-    function getVestingSchedule(bytes32 vestingScheduleId)
-        public
-        view
-        returns(VestingSchedule memory){
-        return vestingSchedules[vestingScheduleId];
+     * @dev Computes the releasable amount of tokens for a vesting schedule.
+     * @return the amount of releasable tokens
+     */
+    function _computeReleasableAmount(
+        VestingSchedule storage vestingSchedule
+    ) internal view returns (uint112) {
+        uint256 currentTime = getCurrentTime();
+        unchecked {
+            if (
+                (currentTime < vestingSchedule.cliff) || vestingSchedule.revoked
+            ) {
+                return 0;
+            } else if (
+                currentTime >= vestingSchedule.start + vestingSchedule.duration
+            ) {
+                return vestingSchedule.amountTotal - vestingSchedule.released;
+            } else {
+                // start + duration > currentTime >= cliff >= start
+                uint256 timeFromStart = currentTime - vestingSchedule.start;
+                // duration > timeFromStart >= 0
+                // uint112 * uint32 < uint256, cannot overflow
+                uint256 vestedAmount = (vestingSchedule.amountTotal *
+                    timeFromStart) / vestingSchedule.duration;
+                return uint112(vestedAmount) - vestingSchedule.released;
+            }
+        }
     }
 
     /**
-    * @dev Returns the amount of tokens that can be withdrawn by the owner.
-    * @return the amount of tokens
-    */
-    function getWithdrawableAmount()
-        public
-        view
-        returns(uint256){
-        return _token.balanceOf(address(this)).sub(vestingSchedulesTotalAmount);
+     * @notice Batch create vesting schedules.
+     * @param _vestingSchedules vesting schedules to create
+     */
+    function batchCreateVestingSchedule(
+        VestingSchedule[] calldata _vestingSchedules
+    ) external onlyOwner {
+        unchecked {
+            uint256 _vestingSchedulesTotalAmount = vestingSchedulesTotalAmount;
+            uint256 totalAmount = _vestingSchedulesTotalAmount;
+            uint256 length = _vestingSchedules.length;
+            for (uint256 i = 0; i < length; i++) {
+                VestingSchedule calldata vestingSchedule = _vestingSchedules[i];
+                address beneficiary = vestingSchedule.beneficiary;
+                if (beneficiary == address(0)) revert InvalidBeneficiary();
+                uint112 amount = vestingSchedule.amountTotal;
+                if (amount == 0) revert InvalidAmount();
+                totalAmount += amount;
+                uint32 start = vestingSchedule.start;
+                if (vestingSchedule.cliff < start) revert CliffBeforeStart();
+                uint32 duration = vestingSchedule.duration;
+                if (duration == 0) revert InvalidDuration();
+                require(start + duration > start);
```

Comment:
Unnecessary? `duration` has to be positive at this point and both `duration` and `start` are unsigned. Hence, `start + duration` has to be `> start`.

Reply:
It may overflow.